modules/platform/nuxeo-webengine/nuxeo-webengine-core/src/main/java/org/nuxeo/ecm/webengine/app/JsonWebengineWriter.java

```diff
@@ -102,6 +104,19 @@ protected static void writeException(JsonGenerator jg, Throwable t, MediaType me
         jg.flush();
     }
 
+    /**
+     * @since 11.5
+     */
+    protected static String getExceptionMessage(String exceptionMessage, int statusCode) {
+        if (statusCode < SC_INTERNAL_SERVER_ERROR) {
+            return exceptionMessage;
+        }
+
+        boolean stackDisplay = jsonFactoryManager.isStackDisplay();
+        boolean isDevMode = Framework.isDevModeSet();
+        return stackDisplay || isDevMode ? exceptionMessage : "Internal Server Error";
```

Comment:
Why not simply

    if (statusCode < SC_INTERNAL_SERVER_ERROR || jsonFactoryManager.isStackDisplay() || Framework.isDevModeSet()) {
        return exceptionMessage;
    } else {
        return "Internal Server Error";
    }